packages/sms/src/rest/v1/sms-domain-api.ts

```diff
@@ -67,9 +67,8 @@ export class SmsDomainApi implements Api {
     if (!this.client) {
       const region = this.sinchClientParameters.region || Region.UNITED_STATES;
       const apiClientOptions = this.buildApiClientOptions(this.sinchClientParameters, region);
-      const basePath = `https://${apiClientOptions.useServicePlanId?'':'zt.'}${region}.sms.api.sinch.com`;
       this.client = new ApiFetchClient(apiClientOptions);
-      this.client.apiClientOptions.basePath = basePath;
+      this.client.apiClientOptions.hostname = this.sinchClientParameters.smsHostname ?? `https://${apiClientOptions.useServicePlanId?'':'zt.'}${region}.sms.api.sinch.com`;
```

Comment:
Same remark as for conversation: design do not enable to set a dynamic domain and use SDK logic to handle regions
Could require a note for SDK's user (_region is not used when hostname is_)